index.ts

```diff
@@ -0,0 +1,217 @@
+            // 4. Read beacon
+            // **************************************************************************
+            const beaconResponse = await rrpBeaconServer
+              .connect(airnodeWallet)
+              .readBeacon(templateId);
+            // const beaconResponse = { value: ethers.BigNumber.from("683392028") };
+
+            if (!beaconResponse) {
+              console.log("Error: failed to fetch data from beacon server");
+              return;
+            }
+            console.log("Info: beacon server value", beaconResponse.value);
+
+            // **************************************************************************
+            // 5. Check deviation
+            // **************************************************************************
+            const delta = beaconResponse.value.sub(apiValue).abs();
+            if (delta.eq(0)) {
+              console.log("Info: beacon is up-to-date. skipping update");
+              return;
+            }
+
+            const deviation = delta
+              .mul(100 * Number(reservedParameters._times)) // TODO: can _times be null or 0?
+              .div(apiValue);
+            console.log(
+              "Info: deviation %",
+              deviation.toNumber() / Number(reservedParameters._times)
+            );
+
+            // **************************************************************************
+            // 6. Update beacon if necessary (call makeRequest)
+            // **************************************************************************
+
+            // TODO: should we calculate the requestId hash or find a way to prevent sending
+            // the same request more that once? RrpBeaconServer.requestIdToTemplateId keeps
+            // track of the pending requests using a templateId
+
+            // TODO: 5% is hardcoded, should this be read from config?
+            const tolerance = 5;
+            if (deviation.lte(tolerance * Number(reservedParameters._times))) {
+              console.log(
+                "Info: delta between beacon and api value is within tolerance range. skipping update"
+              );
+              return;
+            }
+
+            /*
+             * 1. Airnode must first call setSponsorshipStatus(rrpBeaconServer.address, true) to
+             *    enable the beacon server to make requests to AirnodeRrp
+             * 2. Sponsor should then call setUpdatePermissionStatus(airnodeWallet.address, true)
+             *    to allow requester to update beacon
+             */
+            // console.log(
+            //   "🚀 ~ file: index.ts ~ line 161 ~ handler ~ await rrpBeaconServer.sponsorToUpdateRequesterToPermissionStatus()",
+            //   await rrpBeaconServer.sponsorToUpdateRequesterToPermissionStatus(
+            //     airnodeWallet.address,
+            //     airnodeWallet.address
+            //   )
+            // );
+
+            // TODO: who should be the sponsor?
```

Comment:
I suppose this is another config file thing, `sponsorWalletPath` set per job

Reply:
So we would enter `m/123/0`, `m/123/1`, `m/123/2` etc. for different jobs (or the same path for all the jobs)

Reply:
I'm a bit confused about this `sponsorWalletPath` configuration. If the idea is that we `rrpBeaconServer.requestBeaconUpdate()` more than once with different sponsor wallets shouldn't we use a a sponsor address array instead per job?

Something like:

```
    "rrpBeaconServerKeeperJobs": [
      {
        "templateId": "0x50c604914d8ed35473149457a1a0912b785813b4e2e51bd2b75409ca25c50e1d",
        "oisTitle": "Currency Converter API",
        "endpointName": "convertToUSD",
        "deviationPercentage": "5",
        "sponsors": ["0x1234..., 0x5678, etc"]
      }
    ]
```

And Airkeeper would then derive the sponsor wallet for each sponsor in the `rrpBeaconServerKeeperJobs.sponsors` array and `call rrpBeaconServer.requestBeaconUpdate()`

Reply:
The idea is that if I'm a sponsor, I don't want my (RRP) sponsor wallet to be used to request beacon updates because that wallet is already being used to fulfill RRP requests. So the idea here is that I specify the path of the wallet that I want the request updates to be made with.
btw don't confuse this with the sponsor/sponsorWallet parameters that the keeper should use while creating the request